src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,266 @@
+package duke;
+
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.InsertCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.exceptions.InsufficientArgumentsException;
+import duke.exceptions.UnknownCommandException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+
+/**
+ * Represents the Application object responsible for storing and executing commands.
+ */
+public class Parser {
+    public static final String OUTPUT_DATE_TIME_PATTERN = "d MMM yyyy, HHmm";
+    public static final String STORAGE_DATE_TIME_PATTERN = "yyyy-MM-dd HHmm";
```

Comment:
I also like how you organised the pattern in this manner, but would it be better for them to have a common prefix as well? (Eg: DATE_TIME_PATTERN)